app/uk/gov/hmrc/tradergoodsprofiles/controllers/GetRecordsController.scala

```diff
@@ -57,10 +58,23 @@ class GetRecordsController @Inject() (
   ): Action[AnyContent] =
     (authAction(eori) andThen validateHeaderAction).async { implicit request =>
       (for {
+        _      <- validateQueryParameterLastUpdatedDate(lastUpdatedDate)
         record <- routerService.getRecords(eori, lastUpdatedDate, page, size)
       } yield Ok(Json.toJson(record))).merge
     }
 
+  private def validateQueryParameterLastUpdatedDate(lastUpdatedDate: Option[String]): EitherT[Future, Result, Instant] =
```

Comment:
This function does not do what we want, it will wrongly returns a current date (Instant.now()) if the lastUpdateDate is empty, so this will undo what the client is sending in the request
 
Also it is confusing as it returns  (Instant.now()) (if this is empty) and then do not use (we use the empty value anyway), so why bother return Instant.now if this is empty?

Try the below refactor function as I thinks it is more clear and reads better.

`

private def validateQueryParameterLastUpdatedDateS(
    lastUpdateDate: Option[String]
  ): EitherT[Future, Result, Option[Instant]] =
    EitherT.fromEither(
      Try(lastUpdateDate.map(Instant.parse(_))).toEither.left.map(_ =>
        BadRequestErrorsResponse(
          uuidService.uuid,
          Some(Seq(Error(InvalidRequestParameter, InvalidLastUpdatedDate, InvalidLastUpdatedDateCode)))
        ).toResult
      )
    )

The above will: 
1. return an Option[Instant] = None if lastUpdateDate was not added by the client,
2. throw an error this was on the wrong format